### DISCUSSION TITLE
2024-06-28-paiza-c151

### DISCUSSION THREAD

**Discussion by giscus** [ouharu/ouharu.github.io#11] [Show and tell]
# 2024-06-28-paiza-c151

题目：https://paiza.jp/career/challenges/745/retry  题目原题要求对给出的正整数和位数，进行四舍五入计算： 12345入力例1583271 // 正整数3      // 位数出力例1583000   本例中因为从右往左数第3位为2，则四舍五入后结果后3位舍去。 💭思路由此想到可以对正整数除以10^(位数)的商进行四舍五入后再乘回去得到结果 1234

https://ouharu.github.io/2024-06-28-paiza-c151

**Comment by ouharu** [comment]
输出小数位数的精度问题：
- 原因：
保留一位小数，如果 `res` 的真实值为 81.19999999,那么上面的代码输出将是 `81.2`。但如果预期的输出应该是 `81.1`,就会出现问题。
-解决：floor(res*10)/10 = floor(811.9999)/10=811/10=81.1

**Comment by ouharu** [comment]
格式化输出保留小数位数，也使用了四舍五入，所以也会带来精度问题。
https://blog.51cto.com/aiyc/5342604

**Comment by ouharu** [comment]
floor(num+0.5)拓展

floor(num*10^n+0.5)/10^n
